Sources/Sentry/include/SentryReplayRecording.h

```diff
@@ -0,0 +1,46 @@
+#import "SentrySerializable.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class SentryId;
+
+static NSString *const SentryReplayEncoding = @"h264";
+static NSString *const SentryReplayContainer = @"mp4";
+static NSString *const SentryReplayFrameRateType = @"constant";
+
+@interface SentryReplayRecording : NSObject
```

Comment:
so this will also have other events in the future (e.g. breadcrumbs, spans, etc, see here https://github.com/getsentry/sentry-javascript/blob/8af205eba4171462c4b7f1d2d45044c97e748e37/packages/replay/src/types/replayFrame.ts#L207-L241)

You probably don't wanna have their properties all toplevel at the ReplayRecording, I'm kind of modelling the hierarchy in the following way:
- RRWebEvent
  - type
  - timestamp
  - data
  - RRWebMetaEvent
    - href
    - height
    - width
  - RRWebVideoEvent
    - size
    - ....

Reply:
Objc is so verbose that I prefer to have all this properties in the top level, especially because some of this "children" events share some properties.

I believe we will need only 3 more properties, Interactions, Breadcrumbs and Spans.

Reply:
I would also prefer the hierarchy @romtsn pointed out even if it's verbose. I think you could even write those new classes in Swift as they shouldn't have any dependencies.

Reply:
Using Swift would be nice, but programmatically speaking, it does not make sense to split this objects, they wont be used alone anywhere else, and as a con we will need to duplicate some information, like dimentions for example.

Reply:
I'm a bit confused about why they chose inheritance for TS. The duplicated information should be in the base class. The rest of the children's classes inherit from the base class, as pointed out by @romtsn.